File: in_Python/0728 Self Dividing Numbers.py
class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        res = []
        for i in range(left,right+1):
            n = i
            flag = True
            while i != 0:
                cur = i % 10
                if cur == 0:
                    flag = False
                    break
                else:
                    if n % cur != 0:
                        flag = False
                        break
                i /= 10
            if flag:
                res.append(n)
        return res
        

File: in_Java/0728 Self Dividing Numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> res = new ArrayList<>();
        for(int i = left; i <= right; i++){
            if(check(i)) res.add(i);
        }
        return res;
    }
    private boolean check(int n){
        int back = n;
        while(back != 0){
            int t = back % 10;
            if(t == 0) return false;
            if(n % t != 0) return false;
            back /= 10;
        }
        return true;
    }
}